qcodes/instrument/meta/nano_device.py

```diff
@@ -0,0 +1,152 @@
+from typing import List, Dict, Tuple, Any
+
+from qcodes.instrument.parameter import Parameter
+from qcodes.instrument.channel import InstrumentChannel
+from qcodes.station import Station
+
+from qcodes.instrument.meta.meta_instrument import (
+    MetaParameter,
+    MetaInstrument
+)
+
+
+class NanoDeviceParameter(MetaParameter):
+    """
+    Meta for a NanoDevice parameter.
+
+    Args:
+        name: Parameter name
+        endpoints: Endpoints to connect to
+        endpoint_names: Names of endpoints to connect to
+        channel: Optionally set the channel this parameter refers to.
+            Defaults to None.
+    """
+    def __init__(
+        self,
+        name: str,
+        endpoints: Tuple[Parameter],
+        endpoint_names: Tuple[str],
+        channel: InstrumentChannel = None,
+        **kwargs
+    ):
+        self._channel = channel
+        super().__init__(
+            name=name,
+            endpoints=endpoints,
+            endpoint_names=endpoint_names,
+            **kwargs
+        )
+
+    @property
+    def channel(self):
+        return self._channel
+
+    def add_parameter(
+        self,
+        cls,
+        name: str,
+        endpoint: Parameter
+    ) -> Parameter:
+        """Add a sub parameter to this parameter,
+        e.g. my_instrument.param.sub_param
+
+        Args:
+            name: Name of sub parameter
+            endpoint: Endpoint parameter to connect to
+
+        Returns:
+            Parameter: The parameter instance
+        """
+        assert not hasattr(
+            self, name
+        ), f"Duplicate parameter name {name}."
```

Comment:
It is generally a bad idea to throw a bare assertion. I think this should be a KeyError to match the standard parameter